NullPointerException thrown when updating an entity with multiple primary keys

This exception occurs in both 12.16.1 and the latest 13.25.1 versions, I have not tried earlier versions.

## Expected behavior
Expect the value of the col column to be updated to "success"
## Actual behavior
java.lang.NullPointerException: Cannot invoke "io.ebean.bean.EntityBean._ebean_getField(int)" because "bean" is null
### Steps to reproduce
Define the table under postgresql and add test records (the same problem occurs when testing under mysql)
```sql

DROP TABLE IF EXISTS "public"."test_multi_pk_update";
CREATE TABLE "public"."test_multi_pk_update" (
  "p1" varchar(32) COLLATE "pg_catalog"."default" NOT NULL,
  "p2" varchar(32) COLLATE "pg_catalog"."default" NOT NULL,
  "col" varchar(64) COLLATE "pg_catalog"."default"
);

INSERT INTO "public"."test_multi_pk_update" VALUES ('p1', 'p2', 'column_content');


ALTER TABLE "public"."test_multi_pk_update" ADD CONSTRAINT "test_id_class_pkey" PRIMARY KEY ("p1", "p2");
```
Define java beans
```java
@Embeddable
public class TestMultiPkUpdateId {
    @AttributeOverride(name = "p1", column = @Column(name = "p1"))
    @Column(name = "p1")
    private String p1;

    @AttributeOverride(name = "p2", column = @Column(name = "p2"))
    @Column(name = "p2")
    private String p2;

    public String getP1() {
        return this.p1;
    }

    public void setP1(String p1) {
        this.p1 = p1;
    }

    public String getP2() {
        return this.p2;
    }

    public void setP2(String p2) {
        this.p2 = p2;
    }

    @Override
    public int hashCode() {
        int result = 17 * 31;
        result = 31 * result + Objects.hashCode(p1);
        result = 31 * result + Objects.hashCode(p2);
        return result;
    }

    @Override
    public boolean equals(Object that) {
        if (this == that) {
            return true;
        }
        if (!(that instanceof TestMultiPkUpdateId other)) {
            return false;
        }
        if (!Objects.equals(p1, other.p1)) return false;
        return Objects.equals(p2, other.p2);
    }
}
```
```java
@Entity(name = "TestMultiPkUpdate")
@Table(name = "test_multi_pk_update")
@IdClass(TestMultiPkUpdateId.class)
public class TestMultiPkUpdate {
    @Id
    @Column(name = "p1")
    private String p1;

    @Id
    @Column(name = "p2")
    private String p2;

    @Column(name = "col", length = 64, nullable = true)
    private String col;

    public String getP1() {
        return this.p1;
    }

    public void setP1(String p1) {
        this.p1 = p1;
    }

    public String getP2() {
        return this.p2;
    }

    public void setP2(String p2) {
        this.p2 = p2;
    }

    public String getCol() {
        return this.col;
    }

    public void setCol(String col) {
        this.col = col;
    }
}
```

Execute the following code to reproduce the exception:
```java
TestMultiPkUpdate testMultiPk = new TestMultiPkUpdate();
        testMultiPk.setP1("p1");
        testMultiPk.setP2("p2");
        testMultiPk.setCol("success");
        // Expect the value of the col column to be updated to "success". And the actual exception is thrown:
        // java.lang.NullPointerException: Cannot invoke "io.ebean.bean.EntityBean._ebean_getField(int)" because "bean" is null
        DB.update(testMultiPk);

        // Error: ERROR: duplicate key value violates unique constraint "test_id_class_pkey"
        // Detailed: Key (p1, p2)=(p1, p2) already exists.
        DB.save(testMultiPk);
```

```console
  Caused by: java.lang.NullPointerException: Cannot invoke "io.ebean.bean.EntityBean._ebean_getField(int)" because "bean" is null
	at io.ebeaninternal.server.properties.EnhanceBeanPropertyAccess$Getter.get(EnhanceBeanPropertyAccess.java:57)
	at io.ebeaninternal.server.deploy.BeanProperty.getValue(BeanProperty.java:768)
```

--- a/ebean-core/src/main/java/io/ebeaninternal/server/persist/dmlbind/BindableIdEmbedded.java
+++ b/ebean-core/src/main/java/io/ebeaninternal/server/persist/dmlbind/BindableIdEmbedded.java
@@ -60,6 +60,10 @@ final class BindableIdEmbedded implements BindableId {
   @Override
   public void dmlBind(BindableRequest request, EntityBean bean) throws SQLException {
     EntityBean idValue = (EntityBean) embId.getValue(bean);
+    if (idValue == null && matches != null) {
+      // The id (e.g. IdClass) hasn't been derived/cached on this bean yet
+      idValue = deriveId(bean);
+    }
     for (BeanProperty prop : props) {
       Object value = prop.getValue(idValue);
       request.bind(value, prop);
@@ -83,13 +87,21 @@ final class BindableIdEmbedded implements BindableId {
 
   @Override
   public boolean deriveConcatenatedId(PersistRequestBean<?> persist) {
+    deriveId(persist.entityBean());
+    return true;
+  }
+
+  /**
+   * Derive/build the concatenated id (e.g. IdClass) from the entity's own matching id
+   * properties, caching it on the bean (via setValueIntercept) for subsequent use.
+   */
+  private EntityBean deriveId(EntityBean bean) {
     if (matches == null) {
       String m = "No matches for " + embId.fullName() + " the concatenated key columns where not found?"
         + " I expect that the concatenated key was null, and this bean does"
         + " not have ManyToOne assoc beans matching the primary key columns?";
       throw new PersistenceException(m);
     }
-    EntityBean bean = persist.entityBean();
     // create the new id
     EntityBean newId = (EntityBean) embId.createEmbeddedId();
     // populate it from the assoc one id values...
@@ -97,7 +109,7 @@ final class BindableIdEmbedded implements BindableId {
       match.populate(bean, newId);
     }
     embId.setValueIntercept(bean, newId);
-    return true;
+    return newId;
   }
 
 }

--- a/ebean-test/src/test/java/org/tests/model/bridge/TestIdClassScalar.java
+++ b/ebean-test/src/test/java/org/tests/model/bridge/TestIdClassScalar.java
@@ -219,4 +219,36 @@ class TestIdClassScalar extends BaseTestCase {
     DB.deleteAll(Arrays.asList(user, site));
   }
 
+  /**
+   * Test for #3275 - update/delete on a fresh bean instance that has not been read
+   * from the database or previously inserted via this same bean instance (e.g. built
+   * directly from external data) used to throw a NullPointerException as the IdClass
+   * id had not yet been derived/cached on the bean.
+   */
+  @Test
+  void update_freshBeanInstance_notPreviouslyLoaded() {
+    UUID siteId = UUID.randomUUID();
+    UUID userId = UUID.randomUUID();
+
+    BSiteUserD access = new BSiteUserD(BAccessLevel.ONE, siteId, userId);
+    DB.save(access);
+
+    // a fresh bean instance - never read from the db or saved via this instance
+    BSiteUserD freshUpdate = new BSiteUserD(BAccessLevel.TWO, siteId, userId);
+    freshUpdate.setVersion(access.getVersion());
+    DB.update(freshUpdate);
+
+    BEmbId id = new BEmbId(siteId, userId);
+    BSiteUserD found = DB.find(BSiteUserD.class, id);
+    assertThat(found).isNotNull();
+    assertThat(found.getAccessLevel()).isEqualTo(BAccessLevel.TWO);
+
+    // a fresh bean instance used for delete
+    BSiteUserD freshDelete = new BSiteUserD(BAccessLevel.TWO, siteId, userId);
+    freshDelete.setVersion(found.getVersion());
+    DB.delete(freshDelete);
+
+    assertThat(DB.find(BSiteUserD.class, id)).isNull();
+  }
+
 }

